src/MJanssen/Provider/RoutingServiceProvider.php

```diff
@@ -49,97 +50,121 @@ public function boot(Application $app)
     }
 
     /**
+     * Adds all routes
+     *
      * @param Application $app
      * @param $routes
      */
     public function addRoutes(Application $app, $routes)
     {
-        foreach ($routes as $route) {
+        foreach ($routes as $routeName => $route) {
+            if(is_string($routeName) && !array_key_exists('routeName', $route)){
+                $route['routeName'] = $routeName;
+            }
             $this->addRoute($app, $route);
         }
     }
 
     /**
+     * Adds a route, a given route-name (for named routes) and all of its methods
+     *
      * @param Application $app
-     * @param $route
+     * @param array $route
      * @throws InvalidArgumentException
      */
     public function addRoute(Application $app, $route)
     {
+
         $this->validateRoute($route);
 
-        foreach($route['method'] AS $method) {
-            $newRoute = $route;
-            $newRoute['method'] = $method;
-            $this->addRouteByMethod($app, $newRoute);
+        $controller = $app->match(
+            $route['pattern'],
+            $route['controller'])
+            ->bind(
+                ($route['routeName']) ? $route['routeName'] : ''
+            )->method(
+                join('|',array_map('strtoupper', $route['method']))
+            );
+
+        $supportedProperties = array('value', 'assert', 'before', 'after');
+        foreach ($supportedProperties AS $property) {
+            if (isset($route[$property])) {
+                $this->addActions($controller, $route[$property], $property);
+            }
+        }
+
+        if (isset($route['scheme'])) {
+            if ('https' === $route['scheme']) {
+                $controller->requireHttps();
+            }
+        }
+    }
+
+    /**
+     * Validates the given methods. Only get, put, post, delete, options, head
+     * are allowed
+     *
+     * @param array $methods
+     */
+    protected function validateMethods(Array $methods)
+    {
+        $availableMethods = array('get', 'put', 'post', 'delete', 'options', 'head');
+        foreach(array_map('strtolower', $methods) as $method){
+            if(!in_array($method, $availableMethods)){
+                throw new InvalidArgumentException('Method "' . $method . '" is not valid, only the following methods are allowed: get, put, post, delete, options, head');
+            }
         }
     }
 
     /**
+     * Validates the given $route Array
+     *
      * @param $route
      * @throws \InvalidArgumentException
      */
-    protected function validateRoute($route)
+    protected function validateRoute(&$route)
```

Comment:
This is passed by reference only because of the sanitation, which doesn't belong in this method anyway (validation is something entirely different then sanitation).

Could you move the sanitation method and also the pass by reference?